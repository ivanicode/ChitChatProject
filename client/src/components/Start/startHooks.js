import {useHistory} from 'react-router-dom';

export function useStartHooks(){
    const {
        redirectToLogin, 
        redirectToRegister
    } = useRedirect()

    return {
        redirectToLogin,
        redirectToRegister
        }
}

export function useRedirect(){
    const history = useHistory();
    function redirectToLogin(){
        history.push('/login')
    }
    function redirectToRegister(){
        history.push('/register')
    }
    return {redirectToLogin, redirectToRegister}
}
