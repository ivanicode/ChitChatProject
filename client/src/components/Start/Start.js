import React from 'react';
import { func } from 'prop-types';
import { useRedirect } from './startHooks';

export function Start(){

    const {
        redirectToLogin, 
        redirectToRegister
    } = useRedirect()

    return (
        <div className="start">
            <div className="logo">
                <img src="/logo.png" alt="logo"></img>
            </div>
            <div className="startButtons">
                <button className="startLogInButton" onClick={redirectToLogin}>Zaloguj się</button>
                <button className="startRegisterButton" onClick={redirectToRegister}>Zarejestruj się</button>
            </div>
        </div>
    )
}
export default Start;
