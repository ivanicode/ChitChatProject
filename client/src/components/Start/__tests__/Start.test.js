import React from 'react';
import {shallow} from 'enzyme';
import {Start} from '../Start';
import * as StartHook from '../startHooks';

describe('Start component', () => {
    it('should match snapshot when no errors', () => {
        StartHook.useStartHooks = jest.fn().mockImplementation(
            () => ({
                redirectToLogin: jest.fn(),
                redirectToRegister: jest.fn()
            })
        )
        const wrapper = shallow(<Start />)
        expect(wrapper).toMatchSnapshot()
    })
    
})