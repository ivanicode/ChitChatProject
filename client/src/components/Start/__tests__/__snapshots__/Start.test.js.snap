// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start component should match snapshot when no errors 1`] = `
<div
  className="start"
>
  <div
    className="logo"
  >
    <img
      alt="logo"
      src="/logo.png"
    />
  </div>
  <div
    className="startButtons"
  >
    <button
      className="startLogInButton"
      onClick={[Function]}
    >
      Zaloguj się
    </button>
    <button
      className="startRegisterButton"
      onClick={[Function]}
    >
      Zarejestruj się
    </button>
  </div>
</div>
`;
